gestalt-module/src/main/java/org/terasology/module/ModuleMetadata.java

```diff
@@ -179,4 +179,30 @@ public void setExtension(String extensionId, Object extension) {
         extensions.put(extensionId, extension);
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, version, displayName, description, requiredPermissions, dependencies, extensions);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
```

Comment:
No need for a null check here, instanceof would allow both the class and null check as one.